tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -245,7 +245,7 @@ final class ScalaJSClassEmitter private (
       js.Block(
           inheritedCtorDef,
           js.Assign(typeVar.prototype, js.New(inheritedCtorRef, Nil)),
-          genAddToPrototype(className, js.Ident("constructor"), typeVar)
+          genAddToPrototype(className, js.StringLiteral("constructor"), typeVar)
```

Comment:
Oh... OK. Strange. So this is more of a workaround than an actual bugfix in Scala.js